src/assets/Generator.Shared/IArmOperationSource.cs

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
+{
+    internal interface IArmOperationSource<T> : IOperationSource<T>
```

Comment:
Only mgmt uses `IOperationSource`, so you can merge base interface into this one.